### PR TITLE
[release-1.25] Update docs to use pkgs.k8s.io instead of legacy repos

### PR DESCRIPTION
**Pull request by xmudrii**
The legacy package repositories (`apt.kubernetes.io` and `yum.kuberentes.io`) are deprecated and frozen. All Kubernetes versions released after September 13, 2023 are published only to the new package repositories hosted at pkgs.k8s.io. Because of that, I think we should update legacy docs to replace legacy repos with the new Kubernetes package repositories.

This is a manual cherry-pick of the followings PRs:

- #42022
- #43407
- #43458
- #43472
- #43478

Note: there might be some other updates included in this PR such as:

- https://github.com/kubernetes/website/pull/43524#discussion_r1362227256

/assign @saschagrunert @cpanato @jeremyrickard 
cc @kubernetes/release-engineering

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by sftim — COMMENTED** [review]
Aside: even for backports, it's handy to add the (unused) shortcode in 1 commit and have a second commit that switches over.

**Comment by xmudrii** [comment]
@sftim Very good point, I updated all four backports

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md [hunk 1]
```diff
@@ -143,73 +144,109 @@ For more information on version skews, see:
...
    sudo apt-get update
    sudo apt-get install -y kubelet kubeadm kubectl
    sudo apt-mark hold kubelet kubeadm kubectl
    ```
+
 {{< note >}}
-In releases older than Debian 12 and Ubuntu 22.04, `/etc/apt/keyrings` does not exist by default.
-You can create this directory if you need to, making it world-readable but writeable only by admins.
+In releases older than Debian 12 and Ubuntu 22.04, `/etc/apt/keyrings` does not exist by default;
+you can create it by running `sudo mkdir -m 755 /etc/apt/keyrings`
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:196]: Why this change?

File: content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+   You should see a single line with the URL that contains your current Kubernetes
+   minor version. For example, if you're using v{{< skew currentVersionAddMinor -1 "." >}},
+   you should see this:
+
+   ```
+   deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v{{< skew currentVersionAddMinor -1 "." >}}/deb/ /
+   ```
+
+1. Change the version in the URL to **the next available minor release**, for example:
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:139]: Is this the right advice for someone who wants to switch repos but not necessarily to upgrade?

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md (hunk 1 above)
Comment (xmudrii) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:196]: I thought the new wording is more explicit and provides actionable instructions how to solve the issue. This is copied/cherry-picked from the main branch.

File: content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md (hunk 2 above)
Comment (xmudrii) [reply content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:139]: Yes, this document is explicitly for upgrading the cluster. Instructions for migrating to the new Kubernetes repositories are outlined in [the announcement blog post](https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/). This blog post is linked from this document in a few different places.

File: content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:196]: My tip: mention the change in the PR description.

(When I see a change that the PR description didn't lead me to expect, I'll often query to check it was intended)

Comment (xmudrii) [reply content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:196]: Good point, updated descriptions for all four PRs

File: content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:139]: - This document doesn't (yet) clearly say it's only relevant when upgrading the cluster.
  - Some people might want to switch package source without upgrading at the same time.
- Patch version upgrades are upgrades, but the the advice to change the minor version would be wrong for someone who plans to upgrade to a newer patch version.

Comment (onlydole) [reply content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:139]: @xmudrii - would you be able to address the previous feedback?

**Comment by onlydole** [comment]
Thank you, @xmudrii! 

/approve
